bin/check_bedfiles.py

```diff
@@ -0,0 +1,107 @@
+	)
+	parser.add_argument(
+	"-l",
+	"--log-level",
+	help="The desired log level (default WARNING).",
+	choices=("CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"),
+	default="WARNING",
+	)
+	return parser.parse_args(argv)
+
+### Function to check if the BED file provided can be used for bcftools and PyRanges
+
+def check_bed_structure(bed_input):
+    colcheck = pd.read_csv(bed_input, sep="\t", header = None)
+    header_test = any(isinstance(value, str) for value in colcheck.iloc[0, 1:2]) ## checks for no full strings in positional argument columns, as chr is interpreted as str
+    if len(colcheck.columns) == 3 and header_test == False:  ### Should contain three columns CHROM, POS, END without header
+        chrom_valid = check_chromosome_col(colcheck.iloc[:,0])
+        start_valid = check_integer_col(colcheck.iloc[:,1])
+        end_valid = check_integer_col(colcheck.iloc[:,2])
+        order_check = check_startend_col(colcheck)
+        if chrom_valid == True: 
+            if start_valid == True and end_valid == True and order_check == True:
+                return True
+            else:
+                return False
+        else:
+            return False
+    else:
+        logger.error("The provided BED file doesn't follow the required format. Please provide a three column BED file (Chromosome, Start, End) without an header. TMB will not be calculated.")
+
+def check_chromosome_col(input):
+    column = input.isin(chroms)
+    column_nonvalid_rows = column.index[column == False].tolist()
+    validity = column.all()
+    if validity == True:
+        logger.info('The Chromosome column is valid.')
+        return True
+    else:
+        logger.error('The Chromosome column contains values not following the "chr1-chr22/X/Y" convention. Please check the rows ', column_nonvalid_rows, ' for problems.')
+        return False
+
+def check_integer_col(input):
+    column = input.map(type).eq(int)
+    validity = column.all()
+    if validity == True:
+        logger.info('The positional argument column only contains integers.')
+        return True
+    else:
+        logger.error('The positional argument column contains non-integer values. Please check your start/end column for problems.')
+        return False
+
+def check_startend_col(input):
+    all_rows_valid = all(input.iloc[:,1] < input.iloc[:,2])
+    if all_rows_valid == False:
+        nonvalid_rows = input[input.iloc[:,1] >= input.iloc[:,2]].index.tolist()
+        logger.error('Not all start positions are smaller than their respective end positions. Please check the rows ', nonvalid_rows, ' for problems.')
+        return False
+    else:
+        return True
+
+
+def main(argv=None):
+	"""Coordinate argument parsing and program execution."""
+	args = parse_args(argv)
+	logging.basicConfig(level=args.log_level, format="[%(levelname)s] %(message)s")
+	if not args.file_in.is_file():
+		logger.error(f"The given input file {args.file_in} was not found!")
```

Comment:
this only logs the error, but continues the program, this should likely just raise an exception instead, see the table https://docs.python.org/3/howto/logging.html